pkg/cluster/connection_pooler.go

```diff
@@ -1060,7 +1097,6 @@ func (c *Cluster) syncConnectionPoolerWorker(oldSpec, newSpec *acidv1.Postgresql
 			if err != nil {
 				return syncReason, err
 			}
-			c.ConnectionPooler[role].Deployment = deployment
```

Comment:
so we change to update internal deployment only if every sync step works, right? Have to think about consequences it might have

Reply:
oh, this is actually a redundant change